File: Data Structures & Algorithms/partition-equal-subset-sum/submission-0.py
class Solution:
    def canPartition(self, nums: List[int]) -> bool:
        summe = sum(nums)
        if summe % 2 != 0:
            return False

        target = summe / 2

        n = len(nums)

        #dp = [[0 for i in range(target)] for j in range(n)]

        dp = [None for i in range(n+1)]
        dp[0] = {0}

        for i in range(1,n+1):
            dp[i] = dp[i-1].copy()
            num = nums[i-1]
            for ps in dp[i-1]:
                dp[i].add(num + ps)

            

        print(dp)

        return target in dp[n]

File: Data Structures & Algorithms/binary-search/submission-0.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l = 0;
        int r = nums.size();
        int mid, mid_val;

        int i = 0;

        while (l<r) {
            mid = l+(r-l) / 2;
            mid_val = nums[mid];
            if (mid_val==target) return mid;
            if (r-l == 1) return -1;
            if (target < mid_val) r = mid;
            else l = mid;
            i++;
            std::cout << "l: " << l << ", mid: "<<mid <<", r: "<<r<<"\n";
            if (i>10) break;
        }

        return -1;
    }
};
